src/metax_api/api/oaipmh/base/metax_oai_server.py

```diff
@@ -302,18 +321,30 @@ def listRecords(self, metadataPrefix=None, set=None, cursor=None, from_=None,
         data = []
         records = self._get_filtered_records(set, cursor, batch_size, from_, until)
         for record in records:
-            data.append(self._get_oai_item(record, metadataPrefix))
+            identifier = self._get_record_identifier(record, set)
+            if set == 'datacatalogs':
+                data.append(self._get_oai_catalog_item(identifier, record, metadataPrefix))
+            else:
+                data.append(self._get_oai_item(identifier, record, metadataPrefix))
         return data
 
     def getRecord(self, metadataPrefix, identifier):
         """Implement OAI-PMH verb GetRecord."""
         try:
-            record = CatalogRecord.objects.get(
-                # data_catalog__catalog_json__identifier__in=self._get_default_set_filter(),
-                identifier__exact=identifier
-            )
+            if CatalogRecord.objects.filter(identifier__exact=identifier).exists():
+                record = CatalogRecord.objects.get(identifier__exact=identifier)
```

Comment:
This pattern (the two above lines) causes two requests to the db - while not fatal here probably, for future reference, the following would be better:
```
try:
    record = CatalogRecord.objects.get(identifier__exact=identifier)
    json = record.research_dataset
    type = 'Dataset'
except CatalogRecord.DoesNotExist:
    # DataCatalog stuff
```
Not necessary, but more a FYI to do with django